repositories/createPost.ts

```diff
@@ -0,0 +1,28 @@
+import * as firebase from "firebase";
+import GameSituation from "../models/GameSituation";
+import { PostTitle, PostBody, PostId } from "../models/Post";
+
+export interface PostData {
+  title: PostTitle;
+  body: PostBody;
+  gameSituation: GameSituation;
+}
+
+export type CreatePost = (postData: PostData) => Promise<PostId>;
+
+export function createCreatePost({
+  firebaseApp
+}: {
+  firebaseApp: firebase.app.App;
+}): CreatePost {
+  return async ({ title, body, gameSituation }: PostData) => {
```

Comment:
I think you don't need specify argument's type here. Because it's defined at line 11

```suggestion
  return async ({ title, body, gameSituation }) => {
```